File: shared-atlas-dart/lib/src/interfaces/create_block_dto.dart
import 'package:shared_atlas_dart/shared_atlas_dart.dart';

class CreateBlockDto {
  final BlockType type;
  final String content;
  final dynamic metadata;
  final String? parentBlockId;
  final num? position;

  CreateBlockDto({
    required this.type,
    required this.content,
    this.metadata,
    this.parentBlockId,
    this.position,
  });

  factory CreateBlockDto.fromJson(Map<String, dynamic> json) {
    return CreateBlockDto(
      type: BlockType.fromJson(json['type']),
      content: json['content'] as String,
      metadata: json['metadata'] as dynamic,
      parentBlockId: json['parentBlockId'] as String?,
      position: json['position'] as num?,
    );
  }
  Map<String, dynamic> toJson() {
    return {
      'type': this.type.toJson(),
      'content': this.content,
      'metadata': this.metadata,
      'parentBlockId': this.parentBlockId,
      'position': this.position,
    };
  }
}

File: shared-atlas-rust/src/interfaces/create_block_dto.rs
use serde::{Serialize, Deserialize};
use crate::interfaces::block_type::BlockType;
use serde_json;

/// Payload for creating a new block.
/// 创建新块的请求体。
/// 
/// API: `POST /api/notes/:noteId/blocks`
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateBlockDto {
/// Type of the block.
/// 块的类型。
    pub r#type: BlockType,
/// Content of the block.
/// 块的内容。
    pub content: String,
/// Optional metadata.
/// 可选的元数据。
    pub metadata: Option<serde_json::Value>,
/// Optional parent block ID.
/// 可选的父块ID。
    pub parent_block_id: Option<String>,
/// Optional position (defaults to end).
/// 可选的位置（默认为末尾）。
    pub position: Option<f64>,
}